File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.py
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        self.grid = grid
        n = len(grid)
        ans = []
        
        for y in range(1, n - 1):
            ans.append([self.get_local_max(x, y) for x in range(1, n - 1)])
                
        return ans
                
    def get_local_max(self, x, y):
        local_max = 0
        
        for j in range(y - 1, y + 2):
            for i in range(x - 1, x + 2):
                local_max = max(local_max, self.grid[j][i])
                
        return local_max

File: leetCode/2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.java
class Solution {
    int[][] grid;
    public int[][] largestLocal(int[][] grid) {
        this.grid = grid;
        int[][] answer = new int[grid.length - 2][grid.length - 2];
        for (int y = 1; y < grid.length - 1; y++) {
            for (int x = 1; x < grid.length -1; x++) {
                answer[y-1][x-1] = getLocalMax(x, y);
            }
        }
        return answer;
    }
    
    private int getLocalMax(int x, int y) {
        int localMax = 0;
        for (int j = y - 1; j < y + 2; j++) {
            for (int i = x - 1; i < x + 2; i++) {
                localMax = Math.max(localMax, this.grid[j][i]);
            }
        }
        return localMax;
    }
}